source/postcard/src/de/flavors.rs

```diff
@@ -116,8 +116,37 @@ pub trait Flavor<'de>: 'de {
         None
     }
 
-    /// Attempt to take the next `ct` bytes from the serialized message
-    fn try_take_n(&mut self, ct: usize) -> Result<&'de [u8]>;
+    /// Attempt to take the next `ct` bytes from the serialized message.
+    ///
+    /// This variant borrows the data from the input for zero-copy deserialization. If zero-copy
+    /// deserialization is not necessary, prefer to use `try_take_n_temp` instead.
+    fn try_take_n_borrowed(&mut self, ct: usize) -> Result<&'de [u8]>;
+
+    /// Attempt to take the next `N` bytes from the serialized message.
+    ///
+    /// This variant copies the data into a fixed-size buffer.
+    fn try_take_n_const<const N: usize>(&mut self) -> Result<[u8; N]> {
```

Comment:
This could end up returning a fairly large value down the stack, I also worry a bit that it could lead to stack overflow.

Is there a reason you prefer this over the existing try_take_n? There already needs to be bounds checking on the input slice, and I'd be surprised if the compiler didn't inline and optimize based on the calling slice's size.

Reply:
Good point. I'll switch to using `try_take_n_temp`.